sunpy/net/hek/tests/test_hek.py

```diff
@@ -228,3 +277,109 @@ def test_missing_times():
+
+    assert result['obs_meanwavel'][0] == 193.0*u.angstrom
+
+
+@pytest.mark.remote_data
+def test_parse_times():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    client = hek.HEKClient()
+    result = client.search(attrs.Time(tstart,tend), attrs.hek.EventType('CH'))
+
+    for idx in range(len(result)):
+        assert isinstance(result['event_starttime'][idx], Time)
+        assert isinstance(result['event_endtime'][idx], Time)
+
+
+@pytest.mark.remote_data
+def test_ssw_latest_events_flares():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    event_type = 'FL'
+    result = Fido.search(attrs.Time(tstart,tend), attrs.hek.EventType(event_type), attrs.hek.FRM.Name == 'SSW Latest Events')
+    assert len(result[0]) == 2
+
+
+@pytest.mark.remote_data
+def test_not_ssw_latest_events_flares():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    event_type = 'FL'
+    result = Fido.search(attrs.Time(tstart,tend), attrs.hek.EventType(event_type), attrs.hek.FRM.Name != 'SSW Latest Events')
+    assert len(result[0]) == 19
+
+
+@pytest.mark.remote_data
+def test_flares_peak_flux():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    event_type = 'FL'
+    result = Fido.search(attrs.Time(tstart,tend), attrs.hek.EventType(event_type), attrs.hek.FL.PeakFlux > 4000.0)
+    assert len(result[0]) == 1
+    for row in result[0]['fl_peakflux']:
+        assert row.value > 4000.0
+
+
+@pytest.mark.remote_data
+def test_flares_peak_flux_and_position():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    event_type = 'FL'
+    result = Fido.search(attrs.Time(tstart,tend), attrs.hek.EventType(event_type), attrs.hek.Event.Coord1 > 800, attrs.hek.FL.PeakFlux > 1000)
+    assert len(result[0]) == 7
+
+    for row in result[0]['event_coord']:
+        assert row.Tx > 800*u.arcsec
+
+    for row in result[0]['fl_peakflux']:
+        assert row.value > 1000.0
+
+
+@pytest.mark.remote_data
+def test_flares_python_logical_ops():
+    tstart = '2011/08/09 07:23:56'
+    tend = '2011/08/09 12:40:29'
+    event_type = 'FL'
+    result = Fido.search(attrs.Time(tstart,tend), attrs.hek.EventType(event_type), (attrs.hek.Event.Coord1 > 50) and (attrs.hek.FL.PeakFlux > 1000))
+    assert len(result[0]) == 7
+
+    for row in result[0]['event_coord']:
+        assert row.Tx > 50*u.arcsec
+
+    for row in result[0]['fl_peakflux']:
+        assert row.value > 1000.0
+
+def test_event_types():
+    tstart = '2017/09/06 11:59:04'
+    tend = '2017/09/06 17:05:04'
+    events = ['AR', 'CE', 'CD', 'CW', 'FI', 'FE', 'FA', 'LP', 'OS', 'SS', 'EF', 'CJ', 'PG', 'OT', 'NR', 'SG', 'SP', 'CR', 'CC', 'ER', 'TO', 'HY', 'BU', 'EE', 'PB', 'PT']
+
+    for event_type in events:
+        # Just to make sure there is no errors happens
+        Fido.search(attrs.Time(tstart,tend), attrs.hek.EventType(event_type))
```

Comment:
This should be parameterized instead using pytest.mark.parametrize for all of the event types.